modelcontextprotocol/tools/dsl.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import json
+from typing import Dict, Any
+
+from client import get_atlan_client
+from pyatlan.model.search import DSL, IndexSearchRequest
+
+# Configure logging
+logger = logging.getLogger(__name__)
+
+
+def get_assets_by_dsl(dsl_query) -> Dict[str, Any]:
+    """
+    Execute the search with the given query
+    Args:
+        dsl_query (Union[str, Dict[str, Any]]): The DSL query as either a string or dictionary
+    Returns:
+        Dict[str, Any]: A dictionary containing the results and aggregations
+    """
+    logger.info("Starting DSL-based asset search")
+    if logger.isEnabledFor(logging.DEBUG):
+        logger.debug("Processing DSL query")
+    try:
+        # Convert dictionary to string if needed
+        if isinstance(dsl_query, dict):
+            logger.debug("Converting DSL dictionary to JSON string")
+            dsl_query = json.dumps(dsl_query)
```

Comment:
We might be creating and a dictionary just to dump it later on to again create the dictionary